apps/namada-interface/src/slices/accounts.ts

```diff
@@ -135,3 +143,55 @@ const { actions, reducer } = accountsSlice;
 
 export const { addAccounts } = actions;
 export default reducer;
+
+////////////////////////////////////////////////////////////////////////////////
+// JOTAI
+////////////////////////////////////////////////////////////////////////////////
+
+const accountsAtom = (() => {
+  const base = atom(new Promise<readonly AccountDetails[]>(() => {}));
+
+  return atom(
+    (get) => get(base),
+    async (_get, set) => {
+      const namada = getIntegration("namada");
+      const accounts = namada.accounts().then(async (accounts) => {
```

Comment:
Is there any particular reason here to use then instead of await?

Reply:
In this case it was because I want to reject the accounts promise if accounts is undefined, but I think if I throw an error outside of the `then` callback, it will reject the setter instead. I'm not sure if there is a nicer syntax using await though. Maybe something like:

```
const accounts = (async () => {
  const result = await namada.accounts();
  if (results === undefined) {
    throw new Error("accounts was undefined!");
  }
 return result;
})();

set(base, accounts);
```